opentelemetry/src/baggage.rs

```diff
@@ -282,10 +282,29 @@ impl FromIterator<KeyValueMetadata> for Baggage {
     }
 }
 
+fn encode(s: Cow<'_, str>) -> String {
+    let special_characters = ['.', '-', '_', '~'];
+    let mut encoded_string = String::new();
```

Comment:
nit - we can preallocate the encoded string with the required length `s.len()`?

Reply:
yep